lib/krb5/build_ap_req.c

```diff
@@ -52,14 +52,8 @@ krb5_build_ap_req (krb5_context context,
   ap.ap_options.use_session_key = (ap_options & AP_OPTS_USE_SESSION_KEY) > 0;
   ap.ap_options.mutual_required = (ap_options & AP_OPTS_MUTUAL_REQUIRED) > 0;
 
-  ap.ticket.tkt_vno = 5;
-  copy_Realm(&cred->server->realm, &ap.ticket.realm);
-  copy_PrincipalName(&cred->server->name, &ap.ticket.sname);
-
   decode_Ticket(cred->ticket.data, cred->ticket.length, &t, &len);
```

Comment:
Let's get rid of the `t` local variable and decode directly into `&ap.ticket`.  I checked, and that's [what MIT does](https://github.com/krb5/krb5/blob/master/src/lib/krb5/krb/mk_req_ext.c#L142).